[DOC] Update Traces Drilldown doc for GA (#434)
grafana/traces-drilldown: docs/sources/investigate/add-filters.md

---
description: Investigate trends and spikes to identify issues.
canonical: https://grafana.com/docs/grafana/latest/explore/simplified-exploration/traces/investigate/
keywords:
  - Traces Drilldown
  - Investigate
title: Add filters
menuTitle: Add filters
weight: 600
---

# Add filters to refine your investigation

Use filters to refine your investigation.

Filters are available on the **Breakdown** and **Comparison** views.
Refer to [Analyze tracing data](analyze-tracing-data) for how to use these views.

Each time you add a filter, the condition appears in the list of filters at the top of the page.
The list of filters expands as you investigate and explore your tracing data using Traces Drilldown.

1. Refine your investigation by adding filters.

> 1. Optional: Choose one of the attributes to group by or use **Search** to locate the service.

Reviewer: What does it mean:
> Choose one of the attributes **to group by**

?

Is this still describing the filter or is it describing the **Breakdown** tab?
If it's describing the **Breakdown** tab, isn't it the same as the next point?
> Use the tabs underneath the metrics selection to provide insights into breakdowns, comparisons, latency, and other explorations.

Author: Yes. I've removed that. It was a left over. Glad you read through this!

1. Optional: Use the tabs underneath the metrics selection to provide insights into breakdowns, comparisons, latency, and other explorations.
1. Choose filters to hone in on the problem areas. Each filter that you select adds to the **Filter** bar at the top of the page. You can select filters on the **Comparison** and **Breakdown** tabs in the following ways:
    * Select **Add to filters**.

>     * Select **Inspect**.

Reviewer: This is a bit unclear, only using **Inspect** doesn't change the filter.

Author: I'll check this.

    * Use the **Filter** bar near the top.

![Change filters for your investigation](/media/docs/explore-traces/traces-drilldown-filters-ga-1.png)

### Example

Let's say that you want to investigate a spike in errored root spans longer than 200ms.

1. Select **Root spans**.
1. Select the **Errored traces** tab.

> 1. In the Filter by labeled values, enter `span:duration`, select greater than (`>`) from the dropdown list, and then enter `200ms`.

Author: ```suggestion
1. In the Filter by labeled values, enter `span:duration`, select greater than (`>`) from the drop-down list, and then enter `200ms`.
```

1. Once the data updates, sort the **Errored traces** table by the **Duration** column.

{{< video-embed src="/media/docs/explore-traces/traces-drilldown-errors-root-span-duration-filter.mp4" >}}

## Modify a filter

Selecting an option for a filter automatically updates the displayed data.
If there are no matches, the app displays a “No data for selected query” message.

To modify an applied filter:

1. Select the filter to modify in the filter bar.

> 1. Select an option from the drop-down list.

Reviewer 2: I've also seen the as `dropdown`. You might want to double check the Style Guide to ensure consistency.

Author: Should be drop-down. I'll make sure it's consistent.

Author: done.

You can also click in the **Filter** bar to add filters using drop-down lists.

## Remove filters

> You can remove all or individual filters.

Reviewer: I think this has changed, in the new filter UI users can only remove filter expressions one at a time by clicking on the **X** next to the expression. There is no **Clear filters** anymore.

Is this right @joey-grafana ?

Reviewer 3: correct, I had mentioned elsewhere that there is no longer a clear filters

Author: Fixing. I missed this one.

To remove a filter, select **Remove filter** (**X**) at the end of the filter you want to remove.

To remove all filters, select **Clear filters** (**X**) from the right side of the filter bar.

> ## Change selected time range

Reviewer 2: I think either `Change the time range` or `Select a time range` is more concise as a header.

Author: ```suggestion
## Change the time range
```

    * Select **Add to filters**.
    * Select **Inspect**.
    * Use the **Filter** bar near the top.

![Change filters for your investigation](/media/docs/explore-traces/traces-drilldown-filters-ga-1.png)

### Example

Let's say that you want to investigate a spike in errored root spans longer than 200ms.

1. Select **Root spans**.
1. Select the **Errored traces** tab.
1. In the Filter by labeled values, enter `span:duration`, select greater than (`>`) from the drop-down list, and then enter `200ms`.
1. Once the data updates, sort the **Errored traces** table by the **Duration** column.

{{< video-embed src="/media/docs/explore-traces/traces-drilldown-errors-root-span-duration-filter.mp4" >}}

## Modify a filter

Selecting an option for a filter automatically updates the displayed data.
If there are no matches, the app displays a “No data for selected query” message.

To modify an applied filter:

1. Select the filter to modify in the filter bar.
1. Select an option from the drop-down list.

You can also click in the **Filter** bar to add filters using drop-down lists.

## Remove filters

> You can remove all or individual filters.

Author: ```suggestion
```

To remove a filter, select **Remove filter** (**X**) at the end of the filter you want to remove.

> 

Author: ```suggestion
```

> To remove all filters, select **Clear filters** (**X**) from the right side of the filter bar.

Author: ```suggestion
```

### Example

Let's say that you want to investigate a spike in errored root spans longer than 200ms.

1. Select **Root spans**.
1. Select the **Errored traces** tab.
1. In the Filter by labeled values, enter `span:duration`, select greater than (`>`) from the dropdown list, and then enter `200ms`.
1. Once the data updates, sort the **Errored traces** table by the **Duration** column.

{{< video-embed src="/media/docs/explore-traces/traces-drilldown-errors-root-span-duration-filter.mp4" >}}

## Modify a filter

Selecting an option for a filter automatically updates the displayed data.
If there are no matches, the app displays a “No data for selected query” message.

To modify an applied filter:

1. Select the filter to modify in the filter bar.
1. Select an option from the drop-down list.

You can also click in the **Filter** bar to add filters using drop-down lists.

## Remove filters

You can remove all or individual filters.

To remove a filter, select **Remove filter** (**X**) at the end of the filter you want to remove.

To remove all filters, select **Clear filters** (**X**) from the right side of the filter bar.

> Selecting **Clear filters** resets your investigation back to the first metric you selected.
> For example, if you selected Errors metrics and **Group by** the `host` service.name, selecting **Clear filters** resets the search back to just **Errors** selected as the metric type.

Author: ```suggestion
```

> For example, if you selected Errors metrics and **Group by** the `host` service.name, selecting **Clear filters** resets the search back to just **Errors** selected as the metric type.

Author: Done.

## Change selected time range

Use the time picker at the top right to modify the data shown in Traces Drilldown.

> You can select a time range of up to 24 hours in duration.
> This time range can be any 24-hour period in your configured trace data retention period.
> The default is 30 days.

Author: ```suggestion
You can select a time range of up to 24 hours in duration.
This time range can be any 24-hour period in your configured trace data retention period.
The default is 30 days.
These values may be different depending on upon your configuration. 
```

> The default is 30 days.

Reviewer: Yes, the 24h window for TraceQL metrics is configurable:
- for Cloud, the default is 24h but tenants can increase this on request
- for OSS Tempo, the default is 3h (configurable): https://grafana.com/docs/tempo/latest/operations/traceql-metrics/?utm_source=chatgpt.com#set-traceql-metrics-query-options

> The default is 30 days.

Reviewer: ![Screenshot 2025-04-23 at 10 55 08 AM](https://github.com/user-attachments/assets/e5ba8879-d869-403b-8fcb-b56dade98c5f)

> For more information about the time range picker, refer to [Use dashboards](https://grafana.com/docs/grafana/latest/dashboards/use-dashboards/#set-dashboard-time-range).

Author: ```suggestion
For more information about the time range picker, refer to [Use dashboards](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/dashboards/use-dashboards/#set-dashboard-time-range).
```
